### ISSUE TITLE
qr-Code-Validierung

### ISSUE THREAD

**Issue by budda85** [OpenOlitor/OpenOlitor#49]
Ich bekomme beim erstellen einer Rechnung oder eines Datenextrakt immer folgende Fehlermeldung:
> Problem beim Aufruf einer Serverfunktion:
> 
>     Der Bericht konnte nicht erzeugt werden:Error: Bei der qr-Code-Validierung wurde festgestellt, dass die IBAN nicht aus der Schweiz oder Liechtenstein stammt

Wir haben eine deutsche IBAN und BIC in der Projektkonfiguration angeben. Ist das damit nicht möglich eine Rechnung oder ein Datenextrakt zu erstellen? Kann man die qr-Code generierung abstellen? Wird das in der Schweiz genutzt oder wofür wird die benötigt?
IBAN und BIC nutzen wir bis erfolgreich für den SEPA-Export.

**Comment by AlwinEgger** [comment]
Das ist wohl so, die eingesetzte Library enthält einen Check und unterstützt wohl nur Schweizer IBAN :-1: 

@mcmpp I guess we need to include a project setting to exclude QR invoice since the workaround of setting a fake Swiss IBAN will not work as the IBAN is used to generate SEPA export, right?

**Comment by budda85** [comment]
Würde es hier einen Workaround geben oder kann ich irgndwie anders eine Rechnung erzeugen?
Wir würden damit gerne Spendenbescheinigungen erstellen.

**Comment by AlwinEgger** [comment]
Wenn du keine Dokumente für deine Bank erzeugen musst (SEPA / Lastschrift), dann gibt es einen Workaround: Du erfasst in den Projekteinstellungen eine gültige (Schweizer) IBAN. Eine solche ist bspw. `CH96 3000 0001 6073 4208 4` (Das Konto gibt es aber nicht.)

**Comment by budda85** [comment]
Das geht, danke.
Man könnte die IBAN auch auf DE, CH oder LI kontrollieren, bevor der QR-Code erstellt wird.
Eventuell ist das einfacher.

Offtopic-Frage:
Bei dem Datenextrakt der Rechnung gibt es es keinen`Vornamen`und `Nachnamen` sondern nur `kunde.bezeichnung`.
Hier ist der Name allerdings als `Nachname Vorname` angegeben. Ist es möglich das zu drehen? Oder ich in einer Office-Vorlage nur den Vorname raus bekomme?

**Comment by AlwinEgger** [comment]
Ja, wir müssen bei der Generierung der Rechnungs-Reports checken ob wir in der Schweiz sind und nur dann die QR-Rechnung einfügen. Mein Vorschlag für dich war ja ein Workaround.

Zur weiteren Frage:
Die Rechnung lautet ja auf einen "Kunden" und nicht auf eine Person. In den Daten sind die Personen, wie du gesehen hast, nicht drin. Es müsste ja auch geraten werden, aus welcher Person eines Kunden die `Vorname` / `Nachname` Info genommen werden soll.
Das `kunde.bezeichnung` Feld kannst du für Kunden mit Anzahl Personen > 1 selbst definieren. Sonst wird automatisch `Nachname Vorname` genommen. Das kann im Moment nicht verändert werden.

**Comment by budda85** [comment]
Sollte nur als Idee dienen, anstellen eines Knopfes in den Einstellungen oder so :-)

So kann ich auf Rechungen mit nur einer Person, die Leute nur mit `Nachname Vorname` ansprechen, richtig?

**Comment by AlwinEgger** [comment]
Fürs erste gibt es schon einen PR: https://github.com/OpenOlitor/openolitor-server/pull/577

Ja, auf der Rechnung kannst du nie Leute nur so anreden.

Eine Erweiterung der Daten wäre machbar. Wir könnten alle Personen einfügen. Aber dann müsstest du dann eben "raten", wer verantwortlich ist.

**Comment by budda85** [comment]
> Fürs erste gibt es schon einen PR: [OpenOlitor/openolitor-server#577](https://github.com/OpenOlitor/openolitor-server/pull/577)

Super das reicht mir. Danke euch.

> Ja, auf der Rechnung kannst du nie Leute nur so anreden.
> Eine Erweiterung der Daten wäre machbar. Wir könnten alle Personen einfügen. Aber dann müsstest du dann eben "raten", wer verantwortlich ist.

Alle Personen müssten da nicht rein, mit würde es reichen, wenn ich `Nachname Vorname` einfach zu `Vorname Nachname` drehen könnte
